dataclients/kubernetes/kube_test.go

```diff
@@ -828,6 +828,61 @@ func TestIngress(t *testing.T) {
 	})
 }
 
+func TestConvertPathRule(t *testing.T) {
```

Comment:
this tests for regression. Wondering if there was a nice way to test the improvement in the change set. E.g extending the test api handler with counter for each path, and check that the counter values are as expected.